elixir/apps/domain/lib/domain/billing.ex

```diff
@@ -36,17 +36,33 @@ defmodule Domain.Billing do
 
   # Limits and Features
 
+  def users_limit_exceeded?(%Accounts.Account{} = account, users_count) do
+    not is_nil(account.limits.users_count) and
+      users_count > account.limits.users_count
+  end
+
   def seats_limit_exceeded?(%Accounts.Account{} = account, active_users_count) do
     not is_nil(account.limits.monthly_active_users_count) and
       active_users_count > account.limits.monthly_active_users_count
   end
 
   def can_create_users?(%Accounts.Account{} = account) do
+    users_count = Actors.count_users_for_account(account)
     active_users_count = Clients.count_1m_active_users_for_account(account)
 
-    Accounts.account_active?(account) and
-      (is_nil(account.limits.monthly_active_users_count) or
-         active_users_count < account.limits.monthly_active_users_count)
+    cond do
+      not Accounts.account_active?(account) ->
+        false
+
+      not is_nil(account.limits.monthly_active_users_count) ->
+        active_users_count < account.limits.monthly_active_users_count
+
+      not is_nil(account.limits.users_count) ->
+        users_count < account.limits.users_count
```

Comment:
Answered in a different comment above.

Reply:
I know this is already merged, so not looking for a change in the code at this point, but just had this rattling in the back of my mind and am curious about this one.

In the answer/example you gave in the other comment, I can understand why you would want to leave the `not is_nil()` check, because in theory you could end up with a `nil` in the comparison check that followed it, but here that's not possible right?  

If `account.limits.users_count` were actually `nil`, then it wouldn't match in the `cond` block correct?

Reply:
cond block doesn't do matching, it just evaluates the condition, think of it as of if block with multiple clauses.

Reply:
Ah, sorry, when I said "match" I wasn't necessarily thinking of pattern matching but rather "evaluates to true and then runs the given code for that condition".

As you mentioned if we think of it as an "if block with multiple clauses", then all an `if` check is looking for is truthy values right? (unless Elixir is doing more under the hood)  If that's true, then I would go back to my original argument/question of we couldn't end up with a `nil` in a given block, because it would never be able to evaluate to a truthy value.  And since we aren't checking both of the values that we're putting in the `>` comparison, we should have roughly the same confidence that what we're comparing is what we want to compare.

Also happy to discuss this in a real-time conversation if you want.  Not trying to drag this out, just genuinely curious.